packages/flutter/lib/src/widgets/app.dart

```diff
@@ -555,11 +559,18 @@ class WidgetsApp extends StatefulWidget {
   /// This list must not null. Its default value is just
   /// `[const Locale('en', 'US')]`.
   ///
-  /// The order of the list matters. By default, if the device's locale doesn't
-  /// exactly match a locale in [supportedLocales] then the first locale in
-  /// [supportedLocales] with a matching [Locale.languageCode] is used. If that
-  /// fails then the first locale in [supportedLocales] is used. The default
-  /// locale resolution algorithm can be overridden with [localeResolutionCallback].
+  /// The order of the list matters. The default locale resolution algorithm, 
+  /// [basicLocaleListResolution], uses a variation of the truncation algorithm where
```

Comment:
"The truncation algorithm"? Do we explain this somewhere? I hope we don't need to.